lib/connect/index.js

```diff
@@ -105,8 +105,13 @@ function connect (brokerUrl, opts) {
   }
 
   if (!protocols[opts.protocol]) {
-    opts.protocol = protocolList.filter(function (key) {
-      return typeof protocols[key] === 'function'
+    const isSecure = (['mqtts', 'wss'].indexOf(opts.protocol) !== -1)
+    opts.protocol = protocolList.filter(function (key, index) {
+      if (isSecure && index % 2 === 0) {
```

Comment:
And the `index % 2` could instead be a check against the insecure protocols array :)